### PR TITLE
Add QuantizedMatmul in QAT

### PR DESCRIPTION
**Pull request by RachelXu7**
### PR types

New features
### PR changes

APIs
### Describe

Support quantized matmul in QAT.

usage:
```python
class MyLayer(Layer):
    def __init__(self):
        self.matmul = paddle.nn.quant.QuantizedMatmul()

    def forward(self, x, y):
        # Before: out = paddle.matmul(x, y)
        out = self.matmul(x, y)
        return out
```

### CODE REVIEW

**Review by yghstill — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]
LGMT
单测执行时间修改

File: python/paddle/nn/quant/quant_layers.py [hunk 1]
```diff
@@ -26,6 +26,8 @@
 from paddle.nn.quant.lsq import FakeQuantActLSQPlus, FakeQuantWeightLSQPlus
 from paddle.utils import unique_name
 
+from ...tensor import linalg
```
Comment (wanghaoshuang) [inline python/paddle/nn/quant/quant_layers.py]: 必须相对引用吗？代码风格最好保持一致

**Review by wanghaoshuang — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]
单测中试用Fluid API遗留问题，后续需统一修改~

**Review by XieYunshen — APPROVED** [review]

**Review by momozi1996 — APPROVED** [review]
LGTM

**Review by yghstill — APPROVED** [review]

**Pull request merged**